### PR TITLE
Updating tomlkit version as we need fix of the dataloss bug during copy() method use on Table object

### PR DESCRIPTION
**Pull request by c2tarun**
#### Which issue(s) does this change fix?

#2923 

#### Why is this change necessary?
We use copy() method when trying to parse tomlconfig. The updated version has a fix which prevents values during copy operation.

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [x] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by aahung — APPROVED** [review]

File: requirements/reproducible-linux.txt [hunk 1]
```diff
@@ -296,9 +296,9 @@ text-unidecode==1.3 \
     --hash=sha256:1311f10e8b895935241623731c2ba64f4c455287888b18189350b67134a822e8 \
     --hash=sha256:bad6603bb14d279193107714b288be206cac565dfa49aa5b105294dd5c4aab93
     # via python-slugify
-tomlkit==0.7.0 \
-    --hash=sha256:6babbd33b17d5c9691896b0e68159215a9387ebfa938aa3ac42f4a4beeb2b831 \
-    --hash=sha256:ac57f29693fab3e309ea789252fcce3061e19110085aa31af5446ca749325618
+tomlkit==0.7.2 \
```
Comment (jfuss) [inline requirements/reproducible-linux.txt:299]: We need to make sure we update our THIRD-PARTY-LICENSES file: https://github.com/aws/aws-sam-cli/blob/develop/installer/assets/THIRD-PARTY-LICENSES#L2490

**Review by jfuss — CHANGES_REQUESTED** [review]
The changes seem reasonable. Should we have some additional integ tests so we can make sure this behavior isn't regressed in the future?

**Comment by c2tarun** [comment]
> The changes seem reasonable. Should we have some additional integ tests so we can make sure this behavior isn't regressed in the future?

Sure, added integ test.

> We need to make sure we update our THIRD-PARTY-LICENSES file: https://github.com/aws/aws-sam-cli/blob/develop/installer/assets/THIRD-PARTY-LICENSES#L2490

I think this file wasn't updated in some time. Updating it now, several updates in this file.

**Comment by jfuss** [comment]
We recently updated the THRID-PARTY-LICENSES in https://github.com/aws/aws-sam-cli/pull/2929/files. Looks like you might not have had an updated version of `develop` given the conflicts.

**Review by jfuss — APPROVED** [review]

**Review by sriram-mv — APPROVED** [review]

**Pull request merged**